### PR TITLE
Update Ubuntu 20.04 with 24.04 for Docker base images

### LINKED ISSUE #40743: Replace Ubuntu 20.04 with 24.04 for Docker base images
**Describe the enhancement:**

Ubuntu 20.04's standard support [ends](https://ubuntu.com/about/release-cycle) in April 2025.  Before then, we should change our Docker base images to use Ubuntu 24.04 instead.

**Describe a specific use case for the enhancement or feature:**

To use a supported base image for our Docker images.

**What is the definition of done?**

Beats' Ubuntu-based Docker images use Ubuntu 24.04

### PR DESCRIPTION
**Pull request by vinit-chauhan**
## Proposed commit message

Update Ubuntu 20.04 with ~22.04~ 24.04 for Docker base images. 

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

- Closes #40743

### CODE REVIEW

**Comment by pierrehilbert** [comment]
run docs-build

**Review by mauri870 — APPROVED** [review]
LGTM, this seems to be the only occurrences of the ubuntu image in elastic-agent. I'm not sure if there are other places we should upgrade this as well.

**Comment by mauri870** [comment]
Adding @andrzej-stencel as reviewer since it is the assignee for the issue.

**Comment by vinit-chauhan** [comment]
> LGTM, this seems to be the only occurrences of the ubuntu image in beats. I'm not sure if there are other places we should upgrade this as well.


AFAIK, In all other beat-specific Dockerfiles we use the `golang` image as the base image. 
We are only using Ubuntu images once to package Linux images for Beats.

**Comment by pierrehilbert** [comment]
run docs-build

**Review by andrzej-stencel — APPROVED** [review]

**Comment by pierrehilbert** [comment]
run docs-build

**Comment by andrzej-stencel** [comment]
The issue was actually [updated](https://github.com/elastic/beats/issues/40743#issuecomment-2372123477) to go from 20.04 straight to 24.04, jumping over 22.04. @vinit-chauhan do you think you can update this pull request to move to 24.04?

**Comment by vinit-chauhan** [comment]
Hey @andrzej-stencel - thanks for mentioning the changed title. I'll update the PR.

File: CHANGELOG.next.asciidoc [hunk 1]
```diff
@@ -227,6 +227,8 @@ https://github.com/elastic/beats/compare/v8.8.1\...main[Check the HEAD diff]
 - When running under Elastic-Agent Kafka output allows dynamic topic in `topic` field {pull}40415[40415]
 - The script processor has a new configuration option that only uses the cached javascript sessions and prevents the creation of new javascript sessions.
 - Update to Go 1.22.7. {pull}41018[41018]
+- Replace Ubuntu 20.04 with 22.04 for Docker base images {issue}40743[40743] {pull}40942[40942]
```
Comment (jlind23) [inline CHANGELOG.next.asciidoc:230]: This has to be updated to 24.04 too! @vinit-chauhan

File: dev-tools/packaging/packages.yml [hunk 2]
```diff
@@ -159,7 +159,7 @@ shared:
   - &docker_spec
     <<: *binary_spec
     extra_vars:
-      from: '--platform=linux/amd64 ubuntu:20.04'
+      from: '--platform=linux/amd64 ubuntu:22.04'
```
Comment (jlind23) [inline dev-tools/packaging/packages.yml:162]:
```suggestion
      from: '--platform=linux/amd64 ubuntu:24.04'
```

File: dev-tools/packaging/packages.yml [hunk 3]
```diff
@@ -172,7 +172,7 @@ shared:
   - &docker_arm_spec
     <<: *docker_spec
     extra_vars:
-      from: '--platform=linux/arm64 ubuntu:20.04'
+      from: '--platform=linux/arm64 ubuntu:22.04'
```
Comment (jlind23) [inline dev-tools/packaging/packages.yml:175]:
```suggestion
      from: '--platform=linux/arm64 ubuntu:24.04'
```

File: CHANGELOG.next.asciidoc (hunk 1 above)
Comment (jlind23) [inline CHANGELOG.next.asciidoc:230]:
```suggestion
- Replace Ubuntu 20.04 with 24.04 for Docker base images {issue}40743[40743] {pull}40942[40942]
```

**Review by jlind23 — COMMENTED** [review]
Some changes are required as stated by @andrzej-stencel

**Comment by vinit-chauhan** [comment]
Thanks @jlind23 - I was about to update the branch. 😄 

I've updated it. Take a look and let me know if there are any changes.

**Review by jlind23 — APPROVED** [review]

**Comment by jlind23** [comment]
buildkite test it

**Comment by jlind23** [comment]
run docs-build

**Review by andrzej-stencel — APPROVED** [review]
Awesome, thanks a lot Vinit!

**Comment by andrzej-stencel** [comment]
Right, the image build fails with the error:

```console
 => ERROR [stage-1 10/12] RUN groupadd --gid 1000 auditbeat                                                                                                0.2s
------
 > [stage-1 10/12] RUN groupadd --gid 1000 auditbeat:
0.192 groupadd: GID '1000' already exists
```

I had the same issue in the upgrade for the Elastic Agent repo: https://github.com/elastic/elastic-agent/pull/5644.

This is because the Ubuntu 24.04 container images now already include user `ubuntu` that has uid and gid 1000, see https://askubuntu.com/questions/1513927/ubuntu-24-04-docker-images-now-includes-user-ubuntu-with-uid-gid-1000 and https://bugs.launchpad.net/cloud-images/+bug/2005129.

I recommend to fix this the same way as [I did it](https://github.com/elastic/elastic-agent/pull/5644/commits/a73fe52d0d8cd5586a260b397dacc9241b5ac968) in my PR: by removing the `ubuntu` user. Can you apply the fix @vinit-chauhan? I presume it would be in this file: [https://github.com/elastic/beats/blob/main/dev-tools/packaging/templates/docker/Dockerfile.tmpl](https://github.com/elastic/beats/blob/e345f285634d1478874a9a0ca8e4aab3611d33d4/dev-tools/packaging/templates/docker/Dockerfile.tmpl).

**Comment by vinit-chauhan** [comment]
Hey @andrzej-stencel - I have updated the code. Let me know if there are any further changes. 

Also, Would you please run the tests?

**Comment by pierrehilbert** [comment]
run docs-build

**Comment by jlind23** [comment]
run docs-build

**Comment by pierrehilbert** [comment]
run docs-build

**Pull request merged**